libusb/hid.c

```diff
@@ -160,11 +160,12 @@ static struct hid_hotplug_context {
 	hidapi_thread_state callback_thread;
 
 	/* This mutex prevents changes to the callback list */
-	pthread_mutex_t cb_mutex;
+	pthread_mutex_t mutex;
 
-	int mutex_ready;
-
-	int thread_running;
+	/* Boolean flags */
+	unsigned char mutex_ready : 1;
+	unsigned char mutex_in_use : 1;
+	unsigned char cb_list_dirty : 1;
```

Comment:
any aprticular reason to use bitfields instead of separate char variables?

Reply:
No reason other than compression. Tried discussing this, got no replies, left as-is. It can be changed to regular chars with no issues.

Reply:
I find it to be unnecessary complexity. We've long passed times when we need to compress things like this to gain any benefit.